### ISSUE TITLE
Fix incompatibilities with .net6

### ISSUE THREAD

**Issue by SasLuca** [rmandvikar/csharp-extensions#5]
.net6 added `IEnumerable.DistinctBy` which conflicts with `EnumerableExtensions.DistinctBy`.
It would be nice to add `#if` statements to exclude such extensions based on the .net version in order to avoid incompatibilities.

Reference: https://learn.microsoft.com/en-us/dotnet/api/System.Linq.Enumerable.DistinctBy?view=net-7.0

**Comment by rmandvikar** [comment]
@SasLuca The current code on `dev` branch (nuget [2.10.0-alpha1](https://github.com/rmandvikar/csharp-extensions/releases/tag/nuget-2.10.0-alpha1)) does have that tweak, and the tests target multiple frameworks, `net6.0`, `net.7.0` among others. This change is also in `main` branch (nuget [2.8.1](https://github.com/rmandvikar/csharp-extensions/releases/tag/nuget-2.8.1)). Could you please try that version and verify? 

https://github.com/rmandvikar/csharp-extensions/blob/ee63bb3827af1d567aa6b37ff4c674b868c37f5e/src/rm.Extensions/EnumerableExtension.cs#L882-L896

**Comment by SasLuca** [comment]
I think there might be an issue here with how nuget works. Does nuget just grab precompiled dlls from it's repos or does it pull the source code and compile it specifically for my project?

Despite my project using .net7, if I goto `rm.Extensions.EnumerableExtensions.DistinctBy` in my IDE (Jetbrains Rider), it decompiles the file from the dll and there are no `#if` clauses, which I assume means it was compiled without targeting .net6+

I also tried this in Visual Studio as a sanity check. I created a new console app with .net7, used nuget to get `v2.10.0-alpha1` of rm.Extensions and pasted some code that triggered the issue on my side and I still get it.

Here is the sample code:
```csharp
using rm.Extensions;
using System.Globalization;

var langs = CultureInfo
    .GetCultures(CultureTypes.NeutralCultures)
    .DistinctBy(it => it.TwoLetterISOLanguageName)
    .Skip(1)
    .ToList();
```

Here is the exact error:
```
CS0121
The call is ambiguous between the following methods or properties: 

'System.Linq.Enumerable.DistinctBy<TSource, TKey>(System.Collections.Generic.IEnumerable<TSource>, System.Func<TSource, TKey>)' 

and 

'rm.Extensions.EnumerableExtension.DistinctBy<T, TKey>(System.Collections.Generic.IEnumerable<T>, System.Func<T, TKey>)'
```

As a workaround for now I created this extension for myself:
```csharp
public static IEnumerable<TSource> Distinct<TSource, TKey>(
    this IEnumerable<TSource> source,
    Func<TSource, TKey> keySelector)
{
    return source.DistinctBy(keySelector, null);
}
```

**Comment by rmandvikar** [comment]
Yup, I see that now. My understanding of multi-targeting was wrong. It makes sense now though. The nuget packages are compiled, and [preprocessor directives](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives) help with conditional compilation, i.e. they are not present in the assemblies. Targeting just `netstandard2.0` for nuget package [typically works](https://learn.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-0)  and is best for maximum compatibility with apps, but not in the case of a method collision. For that, I had to target `net6.0` also. 

I've fixed it in https://github.com/rmandvikar/csharp-extensions/commit/7dd4517d72af84d98613e973f5bdc0e1a19f14c8, and released it in [`2.8.2`](https://github.com/rmandvikar/csharp-extensions/releases/tag/nuget-2.8.2), and [`2.10.1-alpha`](https://github.com/rmandvikar/csharp-extensions/releases/tag/nuget-2.10.1-alpha).

**Comment by SasLuca** [comment]
Yup, can confirm the issue is now fixed, thanks!
